### ISSUE TITLE
RCS layers are lost when changing projection

### ISSUE THREAD

**Issue by james-rae** [fgpv-vpgf/fgpv-vpgf#3118]
Changing projection with an RCS key on the url causes RCS layer not to appear in reprojected map.

Initial traces show some type of problem when attempting to construct the legend item for the RCS layer.

Solution will be added after full trace

**Comment by james-rae** [comment]
During CCP development, we encountered race conditions in the legend when adding layers via the `RZ` API.  As part of the solution to this, the legend started tracking layer blueprints that were added via the API (i.e. not in the original config file) so it knew to wait for them to resolve.  This was particularly obvious using WFS layers, which are slow to create upfront.

However, it appears initial RCS layer loads are also considered added via the API. So we had the following chain of events happen.

1. Initial (empty) legend loads.
2. RCS call resolves. Layer is added (as API layer). Legend tracks its blueprint.
3. User changes projection.
4. Map is bookmarked and destroyed.  Map destruction involves deleting the set of layer blueprints. Config object remains intact; this includes RCS layers that were inserted.  Legend tracking of blueprints was untouched.
5. Bookmark is applied to config. In projection change case, this really only involves resetting the map center point to be in the new projection.
6. Map and legend are recreated.  RCS layer is in the config (no RCS call, no waiting, no API add). During legend creation, the logic sees there is no blueprint (as expected).  But it also sees the RCS layer id in the internal legend blueprint tracker. So it assumes the blueprint is being generated and doesn't issue the command to make one.
7. Legend logic runs with assumption the RCS blueprint exists, it does not, donethanks.

Current patch solution is to have `mapService.destroyMap()` also clear out the legend's internal blueprint tracker. Will be opening another issue to fully pontificate on the finer points of managing layers coming from non-config sources.
